backend/dataline/services/llm_flow/utils.py

```diff
@@ -109,6 +109,29 @@ def from_uri(
         engine = create_engine(database_uri, **_engine_args)
         return cls(engine, schemas=schemas, **kwargs)
 
+    def custom_run_sql_stream(self, query: str) -> Generator[Sequence[Row[Any]], Any, None]:
+        # https://docs.sqlalchemy.org/en/20/core/connections.html#streaming-with-a-fixed-buffer-via-yield-per
+        yield_per = 1000
+        if self.dialect == "mssql":
```

Comment:
This is becoming a little annoying seeing all these if-statements everywhere. Idk if it's time for this yet, cause it's still acceptable, but we should at least start thinking about using some better design pattern here, ex. Command pattern or something where based on the Dialect, the functionality is housed in one place.